keps/sig-windows/2578-windows-conformance/README.md

```diff
@@ -0,0 +1,431 @@
+- Ability to launch hostProcess containers which can run other privileged Windows system API (to be specified further in the future).
+- Ability to utilize NT Authority/Network Service should be able to interact with host network objects
+
+#### Active Directory
+
+- Ability to run a pod as a GSMA User with only be able to access allowed resources
+- Ability to read and write from local storage using ActiveDirectory credentials
+- Observed lack of Ability to read and write from local storage when ActiveDirectory protected resources credentials are not present
+
+#### Network Policies
+
+Note that we do not include UDP, ipv6, in the initial definitions for NetworkPolicy conformance on Windows.  Since NetworkPolicy testing support on
+Windows clusters was just recently added to Kubernetes core, we err on the side of a conservative definition for Windows NetworkPolicy which
+is sufficient to address the needs of *most* legacy network security applications.
+
+- Ability to protect IPv4 pods from acceessing other pods when TCP NetworkPolicys are present that block specific pod connectivity.
+- Ability to protect IPv4 pods from acceessing other pods when TCP NetworkPolicys are present that block specific namespace connectibity.
+
+#### Windows Service Proxying
+
+- Ability to support IPv6 and IPv4 services for accessing internal Linux pods from Windows pods over ClusterIP Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Linux pods from Windows pods over NodePort Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Windows pods from Linux pods over ClusterIP Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Windows pods from Linux pods over NodePort Endpoints.
+
+### Implementation
+
+This section details the implementation of a "tool" which verifies these clusters must allow for the following, differentiated set of verifications, specified above.
+
+- core conformance
+- hostProcess
+- activeDirectory
+- networkPolicy
+- serviceProxy
+
+#### Option 1 - Test tags
+
+One way to implement such a tool is to, specifically, target all available "ginkgo" tests in Kubernetes core itself, and run these one at a time (or concurrently).  Not we *do not* assert that these tests *must* run in parallel in order to comprise a conformance clusters (although this is often done in CI, its generally not a requirement of Kubernetes Conformance for linux clusters, either, since cluster capacity is not related to the functionality of a cluster). As an example, one such implementation of such a Conformance tool might be:
+
+```
+--ginkgo.focus=\\[Conformance\\]|\\[NodeConformance\\]|\\[sig-windows\\]|\\[sig-apps\\].CronJob --ginkgo.skip=\\[LinuxOnly\\]|\\[k8s.io\\].Pods.*should.cap.back-off.at.MaxContainerBackOff.\\[Slow\\]\\[NodeConformance\\]|\\[k8s.io\\].Pods.*should.have.their.auto-restart.back-off.timer.reset.on.image.update.\\[Slow\\]\\[NodeConformance\\]"
```

Comment:
We also run a who suite of slow/serial tests, some of them windows only. We should ensure we included those as well.

Reply:
fair point, unskipped `slow`